### PR TITLE
Intercoviewer

### PR DESCRIPTION
**Pull request by RossStewart**
Server side looking up of an InterCompanyInvoice by DocumentNumber.

### CODE REVIEW

**Comment by lewisrenfrew** [comment]
build is probs failing since the Messagin.Host dockerfile is still using the old dockerhub image. No idea how its passed till now? but merging master will sort it

File: src/Facade/Services/InterCompanyInvoiceService.cs [hunk 1]
```diff
@@ -15,6 +15,17 @@ public InterCompanyInvoiceService(IQueryRepository<InterCompanyInvoice> reposito
             this.repository = repository;
         }
 
+        public IResult<InterCompanyInvoice> GetByDocumentNumber(int id)
+        {
+            var invoice = this.repository.FindBy(i => i.DocumentNumber == id);
```
Comment (lewisrenfrew) [inline src/Facade/Services/InterCompanyInvoiceService.cs:20]: does FindById(documentNumber) not work here?

File: src/Resources/IntercompanyInvoiceResource.cs [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
+        public AddressResource DeliveryAddress { get; set; }
+        public string DocumentType { get; set; }
         public int DocumentNumber { get; set; }
+        public string DocumentDate { get; set; }
+        public int RaisedBy { get; set; }
+        public string ExportOrReturn { get; set; }
+        public int AccountId { get; set; }
+        public string DocumentReference { get; set; }
+        public decimal NetTotal { get; set; }
+        public decimal VATTotal { get; set; }
```
Comment (lewisrenfrew) [inline src/Resources/IntercompanyInvoiceResource.cs:15]: Does stylecop complain about VAT not Vat?

**Review by iaincrawf — APPROVED** [review]
lgtm

**Pull request merged**